src/mono/mono/metadata/gc.c

```diff
@@ -695,6 +695,7 @@ mono_wasm_gc_finalize_notify (void)
 	/* use this if we are going to start the finalizer thread on wasm. */
 	mono_coop_sem_post (&finalizer_sem);
 #else
+	// TODO: is `mono_runtime_do_background_work` wrapped with MONO_ENTER_GC_UNSAFE/MONO_EXIT_GC_UNSAFE ?
```

Comment:
@lambdageek do I need to handle this boundary for wasm ? Or is that callback ok to call from outside as is?